numba/targets/arraymath.py

```diff
@@ -21,6 +21,130 @@
 from numba.typing import signature
 from .arrayobj import make_array, load_item, store_item, _empty_nd_impl
 
+import traceback
+
+from numba.extending import intrinsic
+from numba.errors import RequireConstValue
+
+
+@intrinsic
+def _create_tuple_result_shape(tyctx, shape_list, shape_tuple):
+    """
+    This routine converts shape list where the axis dimension has already
+    been popped to a tuple for indexing of the same size.  The original shape 
+    tuple is also required because it contains a length field at compile time 
+    whereas the shape list does not.
+    """
+
+    # The new tuple's size is one less than the original tuple since axis 
+    # dimension removed.
+    nd = len(shape_tuple) - 1
+    # The return type of this intrinsic is an int tuple of length nd.
+    tupty = types.UniTuple(types.intp, nd)
+    # The function signature for this intrinsic.
+    funtion_sig = tupty(shape_list, shape_tuple)
+
+    def codegen(cgctx, builder, signature, args):
+        lltupty = cgctx.get_value_type(tupty)
+        # Create an empty int tuple.
+        tup = cgutils.get_null_value(lltupty)
+
+        # Get the shape list from the args and we don't need shape tuple.
+        [in_shape, _] = args
```

Comment:
I think this is how Siu did it in his example.  I believe you are right though that it doesn't matter so it is a matter of style?  Personally, I like this style but let me know if you really want it changed and I'll change it.